code/modules/mob/living/carbon/human/human.dm

```diff
@@ -279,8 +279,19 @@
 	spreadFire(AM)
 
 //Added a safety check in case you want to shock a human mob directly through electrocute_act.
-/mob/living/carbon/human/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, safety = 0, override = 0)
-	if(!safety)
+/mob/living/carbon/human/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, safety = 0, override = 0, tesla_shock = 0)
+	if(tesla_shock)
+		var/total_coeff = 1
+		if(gloves)
+			var/obj/item/clothing/gloves/G = gloves
+			if(G.siemens_coefficient <= 0)
+				total_coeff -= 0.5
+		if(wear_suit)
+			var/obj/item/clothing/suit/S = wear_suit
+			if(S.siemens_coefficient <= 0)
+				total_coeff = 0

```

Comment:
I've already told you my thoughts about this, but I'll repeat: I think total immunity just because you're wearing a hardsuit is pretty imba. I'd suggest something like `total_coeff -= 0.9` so you can tank it to a certain extent but still not just stand inside the tesla containment unfazed.